envoy/config/logging.go

```diff
@@ -59,63 +59,61 @@ var (
 	}
 )
 
-// AddLogger adds stdout logging configuration to HTTPManager
-func (e *envoyConfiguration) AddLogger(logFormat string, logFormatTemplate interface{}) error {
-	// Creating stdout logging configuration
-	var accessLogFormatString *core.SubstitutionFormatString
+type AccessLogBuilder struct {
+	al *accesslog.AccessLog
+}
+
+func (a *AccessLogBuilder) Validate() error {
+	return a.al.Validate()
+}
+
+func (a *AccessLogBuilder) GetAccessLog() *accesslog.AccessLog {
+	return a.al
+}
+
+func NewJSONAccessLog(template map[string]string) (*AccessLogBuilder, error) {
```

Comment:
Just out of curiosity, the param for `NewJSONAccessLog` is a map but the param for `NewTextAccessLog` is a string. How come?

Reply:
These 2 methods exist exactly because of this - different accepted values for the log formatting in Envoy.